File: Data Structures & Algorithms/moving-average-from-data-stream/submission-1.py
class MovingAverage:

    def __init__(self, size: int):
        self.data = []
        self.size = size

    def next(self, val: int) -> float:
        self.data.append(val)
        while len(self.data) > self.size:
            self.data.pop(0)
        return float(sum(self.data))/len(self.data)
        


# Your MovingAverage object will be instantiated and called as such:
# obj = MovingAverage(size)
# param_1 = obj.next(val)

File: Data Structures & Algorithms/logger-rate-limiter/submission-1.cpp
#include <unordered_map>

class Logger {
private:
    std::unordered_map<string, int> logs; 
public:
    Logger() {
    }
    bool shouldPrintMessage(int timestamp, string message) {
        bool message_logged = false;
        for (auto const&[key, value] : this->logs) {
            if (message == key) {
                message_logged = true;
            }
        }
        if (message_logged) {
            if (timestamp - this->logs[message] < 10) {
                return false;
            } 
        }
        this->logs[message] = timestamp;
        return true;
    }
};

/**
 * Your Logger object will be instantiated and called as such:
 * Logger* obj = new Logger();
 * bool param_1 = obj->shouldPrintMessage(timestamp,message);
 */
